server/debugsession.go

```diff
@@ -0,0 +1,836 @@
+					Value: "FlowAccount",
+					Type:  "FlowAccount",
+					PresentationHint: dap.VariablePresentationHint{
+						Kind:       "class",
+						Visibility: "public",
+					},
+					VariablesReference: ds.storeVariable(account),
+				}
+				index++
+				responseVariables = append(
+					responseVariables,
+					*variable)
+			}
+
+		default:
+			valueRequested := ds.variables[variableRequested]
+			switch value := valueRequested.(type) {
+
+			case *sdk.Account:
+				storage := inter.SharedState.Config.Storage.GetStorageMap(common.Address(value.Address), "storage", false)
+				responseVariables = ds.convertMembersToDapVariables(inter, storage)
+
+			case interpreter.Value, cadence.Value:
+				responseVariables = ds.convertMembersToDapVariables(inter, value)
+
+			}
+
+		}
+
+		ds.send(&dap.VariablesResponse{
+			Response: newDAPSuccessResponse(request.GetRequest()),
+			Body: dap.VariablesResponseBody{
+				Variables: responseVariables,
+			},
+		})
+	}
+}
+
+func (ds *debugSession) convertValueToDapVariable(name string, value cadence.Value) *dap.Variable {
+	referenceHandle := 0
+	switch value.(type) {
+	case cadence.Dictionary, cadence.Array, cadence.Struct, cadence.Resource:
+		referenceHandle = ds.storeVariable(value)
+	}
+	return &dap.Variable{
+		Name:  name,
+		Value: value.String(),
+		Type:  value.Type().ID(),
+		PresentationHint: dap.VariablePresentationHint{
+			Kind:       "property",
+			Visibility: "private",
+		},
+		VariablesReference: referenceHandle,
+	}
+}
+func (ds *debugSession) storeVariable(value any) int {
+	ds.variableHandleCounter++
+	ds.variables[ds.variableHandleCounter] = value
+	return ds.variableHandleCounter
+}
+
+func (ds *debugSession) convertCadenceValueMembersToDapVariables(cadenceValue cadence.Value) []dap.Variable {
+	members := make([]dap.Variable, 0)
+
+	switch value := cadenceValue.(type) {
+	case cadence.Resource:
+		for i, field := range value.ResourceType.Fields {
+			variable := ds.convertValueToDapVariable(field.Identifier, value.Fields[i])
+			if variable != nil {
+				members = append(
+					members,
+					*variable)
+			}
+		}
+	case cadence.Struct:
+		for i, field := range value.StructType.Fields {
+			variable := ds.convertValueToDapVariable(field.Identifier, value.Fields[i])
+			if variable != nil {
+				members = append(
+					members,
+					*variable)
+			}
+		}
+
+	case cadence.Array:
+		i := 0
+		for _, element := range value.Values {
+			variable := ds.convertValueToDapVariable(fmt.Sprintf("[%d]", i), element)
+			if variable != nil {
+				members = append(
+					members,
+					*variable)
+			}
+			i = i + 1
+		}
+
+	case cadence.Dictionary:
+		for _, pair := range value.Pairs {
+			variable := ds.convertValueToDapVariable(pair.Key.String(), pair.Value)
+			if variable != nil {
+				members = append(
+					members,
+					*variable)
+			}
+		}
+	}
+	return members
+}
+
+func (ds *debugSession) convertMembersToDapVariables(inter *interpreter.Interpreter, someValue any) []dap.Variable {
```

Comment:
Maybe instead of having this "catch-all function", have a separate function per type and call the appropriate

Reply:
Refactored it in [fabd27d](https://github.com/onflow/flow-emulator/pull/294/commits/fabd27d41700b6ed90d28d0c6c0eb1a1245fcae0)